src/main/groovy/io/openliberty/tools/gradle/tasks/InstallLibertyTask.groovy

```diff
@@ -59,25 +59,57 @@ class InstallLibertyTask extends AbstractTask {
         return project.configurations.libertyRuntime
     }
 
+    @Input
+    @Optional
+    String getLibertyRuntimeUrl() {
+        return project.liberty.install.runtimeUrl
+    }
+
+    @Input
+    @Optional
+    String getLibertyLicenseCode() {
+        return project.liberty.install.licenseCode
+    }
+
+    @Input
+    @Optional
+    String getLibertyVersion() {
+        return project.liberty.install.version
+    }
+
+
+    @Input
+    @Optional
+    String getLibertyUsername() {
+        return project.liberty.install.username
+    }
+
+    @Input
+    @Optional
+    String getLibertyPassword() {
+        return project.liberty.install.password
+    }
+
+    @Input
+    @Optional
+    String getLibertyType() {
+        return project.liberty.install.type
+    }
+
+
     @TaskAction
     void install() {
-        if (!isLibertyInstalledAndValid(project)) {
-            def params = buildInstallLibertyMap(project)
-
-            project.ant.taskdef(name: 'installLiberty',
-                                classname: 'io.openliberty.tools.ant.install.InstallLibertyTask',
-                                classpath: project.buildscript.configurations.classpath.asPath)
-            project.ant.installLiberty(params)
-
-            String licenseFilePath = project.configurations.getByName('libertyLicense').getAsPath()
-            if (licenseFilePath) {
-                def command = "java -jar " + licenseFilePath + " --acceptLicense " + project.buildDir
-                def process = command.execute()
-                process.waitFor()
-            }
-            
-        } else {
-            logger.info ("Liberty is already installed at: " + getInstallDir(project))
+        def params = buildInstallLibertyMap(project)
```

Comment:
So why don't we need some check here? This won't get called if it was already installed?

Reply:
Exactly. Even if the Inputs weren't up to date, this check would keep it from reinstalling WLP with the updated configuration. We are already adding a checking if the contents are there in this line
https://github.com/OpenLiberty/ci.gradle/blob/master/src/main/groovy/io/openliberty/tools/gradle/tasks/InstallLibertyTask.groovy#L51